adr/0006-reviewer-responsibilies.adoc

```diff
@@ -0,0 +1,74 @@
+= Reviewer and committer responsibilities
+
+* Status: proposed
+* Date: 2024-10-25
+
+== Context and Problem Statement
+
+Today, we give different roles, responsibilities and resource access across Quarkus contributors. 
+It is not well known nor written down what the review and push responsibilities are and how they ares structured to achieve Quarkus goals of high code quality and velocity.
+
+This leads to different interpretations and some unknown on how contributors can best help.
+At the same time, some contributions remain unreviewed for a long time because of uncertainty or because of bottleneck.
+
+== Decision Drivers
+
+* Increase clarity and awareness on contributor’s responsibility on reviews and code push
+* Describe the intent and not the current exact state (as tooling is preventing us from having the best solution)
+* Being a perfect solution is _not_ a goal, _better_ than today is.
+* Define ways to measure and improve the time to review.
+
+== Scenarios (optional)
+
+List of scenarios, each described using 2-3 sentences, to capture where the ADR has an impact.
+
+== Considered options
+
+The main alternative is the status quo which roughly is: Guillaume knowing what we are intending to do and some persons talking with Guillaume regularly being sufficiently aware.
+
+== Decision
+
+There are three levels of reviewing.
+
+. Any one can review a Pull Request (PR) and give constructive feedback on different areas (code, documentation, feature experience, security, etc.).
+  This is an optional but recommended step.
+  More eyes means higher quality.
+. A PR is reviewed by a _binding reviewer_ who is fluent in the area being touched: this person is responsible for signing with blood of the usefulness, quality of the code and soundness of the solution.
+  That’s the review of the expert and the most critical one for quality.
+  This review is mandatory.
+.. Unfortunately, we do not always have (designed or effective) experts in all areas so it’s a constant work in progress and might make this step optional in some situations.
+. A PR is reviewed by a committer (push rights) whose responsibilities are:
+.. Do a general review (often from a non expert eye).
+.. Ensure consistency and quality across Quarkus’s codebase (code, concepts, API consistency, documentation).
+.. Detect potential security concerns with an attacker mindset (though not being an actual security professional).
+.. Ensure consistency at the platform level (duplicated or overlapping features, usage integration issues across features etc).
+.. Effectively push the code to the main branch.
+
+=== TODO Open questions to resolve before closing
```

Comment:
Why `TODO`?

Reply:
It's really the meat to discuss, we could get more proposals from others. After the conversation, I anticipated "TODO Open questions to resolve before closing" will be folded into the other sections.

Reply:
I got some messages, and Ok, we had this situation. I'm going to improve my tooling to prevent/detect these cases (they were touching areas that my tooling was not monitoring - so it's on me).